src/toil/jobStores/aws/jobStore.py

```diff
@@ -233,7 +233,7 @@ def _registered(self, value):
                                                        attributes=attributes)
     
     def _awsJobFromItem(self, item):
-        if "overlargeID" in item:
+        if item["overlargeID"]:
```

Comment:
Can this potentially KeyError?

Reply:
I wanted this to KeyError because it will fail if overlargeID isn't always present--this makes sure there isn't some sort of logic bug which would cause us to fail to update that key. I could add a catch which raises a more descriptive error, if you'd prefer?

It *will* KeyError when resuming from jobStores created with a previous version, but the other changes throw resumability out the window anyway.